modules/onerepo/src/core/tasks/commands/tasks.ts

```diff
@@ -297,3 +321,19 @@ function slugify(str: string) {
 
 type ExtendedRunSpec = RunSpec & { meta: { name: string; slug: string }; fn?: PromiseFn };
 type TaskList = Array<Array<ExtendedRunSpec>>;
+
+function shardTasks(tasks: TaskList, shard: number, totalShards: number) {
+	if (!tasks.length) {
+		return tasks;
+	}
+	const shardSize = Math.ceil(tasks.length / totalShards);
+	let index = 0;
+	let resIndex = 0;
+
+	const result = new Array(Math.ceil(tasks.length / shardSize));
+
+	while (index < tasks.length) {
+		result[resIndex++] = tasks.slice(index, (index += shardSize));
```

Comment:
does the post increment happen after the assignment? havent seen it done like this inline before, just wanted to double check.

Reply:
I stole this from lodash :shame:

Reply:
gotcha gotcha well it should be reliable then!

Reply:
But yes, `resIndex++` will increment after, `++resIndex` would pre-increment